main/src/ca/uwaterloo/flix/api/Flix.scala

```diff
@@ -522,8 +522,8 @@ class Flix {
       (_, regionErrors) = Regions.run(afterTyper)
       afterEntryPoint <- EntryPoint.run(afterTyper).withSoftFailures(regionErrors)
       _ <- Instances.run(afterEntryPoint, cachedTyperAst, changeSet)
-      afterPredDeps <- PredDeps.run(afterEntryPoint)
-      afterStratifier <- Stratifier.run(afterPredDeps)
+      (afterPredDeps, predDepErrors) = PredDeps.run(afterEntryPoint)
+      afterStratifier <- Stratifier.run(afterPredDeps).withSoftFailures(predDepErrors)
```

Comment:
`predDepErrors` is always `Nil` since `PredDeps` never fails. Any idea in carrying `predDepErrors` around then? Should we ever introduce errors into the phase, it would automatically be picked up by this, but enforcing that it only return `Root` makes it impossible to miss anything. Just thoughts